### PR TITLE
Feature #9523: Display icon or text in vehicle list or in vehicle window indicating whether a vehicle is lost

### PR DESCRIPTION
**Pull request by SamuXarick**
## Motivation / Problem
#9523

## Description
While VF_PATHFINDER_LOST is set for a vehicle, a warning icon is displayed on vehicle window and vehicle list. On the vehicle window, status bar text will change from "heading to" to "can't reach".

Aberhattan Springs Transport, 1935-03-03

## Limitations
I am reusing a sprite! Pretty sure this is not the way to go.

## Checklist for review

Some things are not automated, and forgotten often. This list is a reminder for the reviewers.
* The bug fix is important enough to be backported? (label: 'backport requested')
* This PR touches english.txt or translations? Check the [guidelines](https://github.com/OpenTTD/OpenTTD/blob/master/docs/eints.md)
* This PR affects the save game format? (label 'savegame upgrade')
* This PR affects the GS/AI API? (label 'needs review: Script API')
    * ai_changelog.hpp, gs_changelog.hpp need updating.
    * The compatibility wrappers (compat_*.nut) need updating.
* This PR affects the NewGRF API? (label 'needs review: NewGRF')
    * newgrf_debug_data.h may need updating.
    * [PR must be added to API tracker](https://wiki.openttd.org/en/Development/NewGRF/Specification%20Status)

### CODE REVIEW

**Comment by glx22** [comment]
Reusing a sprite is totally fine here.

File: src/vehicle_gui.cpp [hunk 1]
```diff
@@ -2732,7 +2736,7 @@ struct VehicleViewWindow : Window {
 		const Vehicle *v = Vehicle::Get(this->window_number);
 		switch (widget) {
 			case WID_VV_START_STOP:
-				size->height = std::max(size->height, std::max(GetSpriteSize(SPR_FLAG_VEH_STOPPED).height, GetSpriteSize(SPR_FLAG_VEH_RUNNING).height) + WD_IMGBTN_TOP + WD_IMGBTN_BOTTOM);
+				size->height = std::max(size->height, std::max(GetSpriteSize(SPR_WARNING_SIGN).height, std::max(GetSpriteSize(SPR_FLAG_VEH_STOPPED).height, GetSpriteSize(SPR_FLAG_VEH_RUNNING).height)) + WD_IMGBTN_TOP + WD_IMGBTN_BOTTOM);
```
Comment (LordAro) [inline src/vehicle_gui.cpp]: `std::max({foo, bar, baz})` using an initializer list would probably be better than chaining std::max calls

File: src/lang/english.txt [hunk 2]
```diff
@@ -4142,6 +4142,11 @@ STR_VEHICLE_STATUS_HEADING_FOR_WAYPOINT_VEL                     :{LTBLUE}Heading
 STR_VEHICLE_STATUS_HEADING_FOR_DEPOT_VEL                        :{ORANGE}Heading for {DEPOT}, {VELOCITY}
 STR_VEHICLE_STATUS_HEADING_FOR_DEPOT_SERVICE_VEL                :{LTBLUE}Service at {DEPOT}, {VELOCITY}
 
+STR_VEHICLE_STATUS_CANT_REACH_STATION_VEL                       :{LTBLUE}Can't reach {STATION}, {VELOCITY}
```
Comment (LordAro) [inline src/lang/english.txt]: There aren't a lot of contractions in the strings at the moment, I wonder whether `Cannot reach...` would be better

**Review by LordAro — COMMENTED** [review]
LGTM, other than a couple of minor points

**Comment by TrueBrain** [comment]
Are you happy with the result? Can we un-draft this? :D

**Comment by SamuXarick** [comment]
It's a bit messed up. It doesn't immediately pick up the status when the route is fixed. And in some cases, it can reach the target while still saying that it cannot be reached. I did a few testings with trains and a missing single track that gets fixed meanwhile, and it took the train some time to clear up the train is lost status.

**Comment by TrueBrain** [comment]
> 
> 
> It's a bit messed up. It doesn't immediately pick up the status when the route is fixed. And in some cases, it can reach the target while still saying that it cannot be reached. I did a few testings with trains and a missing single track that gets fixed meanwhile, and it took the train some time to clear up the train is lost status.

Is this because the GUI doesn't update? Or because the PF just caches information and takes a bit of time to notice the track is restored? The first we can fix. The second .. I personally can accept that as a "known-bug". We cannot instantly update all trains if you put down a single rail :) I think the benefit of this PR out wages the possible questions that might give.

**Comment by SamuXarick** [comment]
The second.

**Comment by TrueBrain** [comment]
*beep beep* IRC to GitHub bot here:

```
<TrueBrain> Samu: not sure what others think, but I fully expected the icon to remain even after the route was fixed .. that is just how OpenTTD works :D
<TrueBrain> the news also remains despite the route being fixed
<TrueBrain> for me that really is not an issue .. but others might disagree :)
<TrueBrain> having some indication is already widely better than nothing, even if it can be there while the route is fixed :)
<FLHerne> TrueBrain: I think it's a bug that the news doesn't vanish too
<FLHerne> but yeah, even with this bug it's a big improvement
<_dp_> train stops being lost when it finds a path, not when tracks are "fixed", train may be on the other side of the map by that point
<_dp_> also would be very helpful if it also showed somewhere exact tile where it became lost
<_dp_> but, yeah, any indication is better than no idication
```

In other words, yes, this PR would "introduce a bug", but not one we might want to solve in this PR :)

**Review by TrueBrain — APPROVED** [review]

**Pull request merged**